### PR TITLE
kakoune: 2020.09.01 -> 2021.08.28

### PR DESCRIPTION
**Pull request by jtrees**
###### Motivation for this change

Bringing kakoune up-to-date.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

###### Additional Notes

ncurses was removed as a dependency upstream:

https://github.com/mawww/kakoune/releases/tag/v2021.08.28

@vrthra

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 136125` at a461d064 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>kakoune</li>
    <li>kakoune-unwrapped</li>
    <li>kakounePlugins.kak-plumb</li>
  </ul>
</details>

File: pkgs/applications/editors/kakoune/default.nix [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
   src = fetchFromGitHub {
     repo = "kakoune";
     owner = "mawww";
     rev = "v${version}";
-    sha256 = "091qzk0qs7hql0q51hix99srgma35mhdnjfd5ncfba1bmc1h8x5i";
+    sha256 = "13kc68vkrzg89khir6ayyxgbnmz16dhippcnw09hhzxivf5ayzpy";
   };
   nativeBuildInputs = [ pkg-config ];
-  buildInputs = [ ncurses asciidoc docbook_xsl libxslt ];
+  buildInputs = [ asciidoc docbook_xsl libxslt ];
```
Comment (Mic92) [inline pkgs/applications/editors/kakoune/default.nix:15]: It also looks like none of those are needed anymore:

```suggestion
```

File: pkgs/applications/editors/kakoune/default.nix [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
   makeFlags = [ "debug=no" ];
 
   postPatch = ''
```
Comment (Mic92) [inline pkgs/applications/editors/kakoune/default.nix:18]: Can you remove the sed patch here? it does not seem to be required.

File: pkgs/applications/editors/kakoune/default.nix (hunk 1 above)
Comment (jtrees) [reply pkgs/applications/editors/kakoune/default.nix]: Oh cool. I don't fully understand what these were there for in the first place (perhaps related to the wrapper that does something with the docs?) but at least it seems to build and execute without.

File: pkgs/applications/editors/kakoune/default.nix (hunk 2 above)
Comment (jtrees) [reply pkgs/applications/editors/kakoune/default.nix]: Nice. Removed.

**Comment by Luflosi** [comment]
This is a duplicate of my older PR #136029 but since there is more progress in this PR, I will close mine.

File: pkgs/applications/editors/kakoune/default.nix [hunk 3]
```diff
@@ -1,25 +1,17 @@
...
-  version = "2020.09.01";
+  version = "2021.08.28";
   src = fetchFromGitHub {
     repo = "kakoune";
     owner = "mawww";
     rev = "v${version}";
-    sha256 = "091qzk0qs7hql0q51hix99srgma35mhdnjfd5ncfba1bmc1h8x5i";
+    sha256 = "13kc68vkrzg89khir6ayyxgbnmz16dhippcnw09hhzxivf5ayzpy";
   };
-  nativeBuildInputs = [ pkg-config ];
```
Comment (Mic92) [inline pkgs/applications/editors/kakoune/default.nix:14]: pkg-config is also no longer required.

File: pkgs/applications/editors/kakoune/default.nix (hunk 1 above)
Comment (Mic92) [reply pkgs/applications/editors/kakoune/default.nix]: They used to be used to build man pages.

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 136125` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>kakoune</li>
    <li>kakoune-unwrapped</li>
    <li>kakounePlugins.kak-plumb</li>
  </ul>
</details>

**Pull request merged**